server/src/main/java/org/elasticsearch/search/dfs/DfsSearchResult.java

```diff
@@ -42,11 +42,22 @@
     private ObjectObjectHashMap<String, CollectionStatistics> fieldStatistics = HppcMaps.newNoNullKeysMap();
     private int maxDoc;
 
-    public DfsSearchResult() {
-    }
-
     public DfsSearchResult(StreamInput in) throws IOException {
-        readFrom(in);
+        super(in);
+        requestId = in.readLong();
```

Comment:
I looked at requestId a bit, since it is kept in SearchPhaseResult, would it make sense to move all reading/writing of it to that super class (any maybe make it final as well?). Currently it looks like a few subclasses that use it write it, others don't. Maybe this can be centralized?

Reply:
Would be nice but we can't make this change, for instance ScrollQuerySearchResult does not set it nor serializes it.

Reply:
I was thinking along the lines of leaving it set to "null" and not serializing/deserializing it as an optional value, but that would mean a break in the serialization format. If you think this generally is a good idea we might better do this as a follow up then?

Reply:
I am not sure that what you suggest would help, I think that it is clearer given that there are differences, if each subclass does its own serialization, also easier to read probably. Also I would not bother breaking the serialization layer for this.